### PR TITLE
Preview variations for emails and landing pages

### PR DESCRIPTION
**Pull request by tejasync**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

As a user I would like to be able to more easily preview emails & landing pages and their variations including:

Translations (if available)
A/B tests
Dynamic content & tokens - [ ]

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create email with at least one AB test and one translation
3. Use at leas in one template tokens like {contactfield=firstname}‍ {contactfield=lastname}‍ to see if contact info is filled properly
4. Test translations and variants preview in email detail page :
106509198-96711a00-64cd-11eb-8d1c-d648c434fe82
5. Use the same scenario for pages

#### Steps to test permissions
Preview actions has now protected access for admins and other roles. Now it is not accessible by anonymous users. There's need to test úit under other role than admin which has no defined resource access right. I've `Sales` role for this.
1. Use admin account and preview s are accessible
2. Use `Sales` or other role defined in mautic with `Email permissions - View Own & View Others` and `Landing page permissions - View Own & View Others` - works with respect to this settings agains user with this role
3. Try to  access previews in anonymous window - this will return access denied

### CODE REVIEW

**Review by kuzmany — DISMISSED** [review]
Looks pretty cool. 
Works as expected. I tested both email and page.
Just If result Is contact without firstname/lastname then return  empty row (probably we should display email in the case)

image

**Comment by RCheesley** [comment]
Thanks for the PR, could you possibly explain how this should work if you have translations AND A/B variants?

I'm not sure if that is possible, but I was trying to preview a Spanish translation with the variant (see screenshot):

screenshot-127 0 0 1_65379-2022 11 23-15_37_29

ID 2 = Root email (English)
ID 4 = A/B variant of email 2
ID 5 = Translation of root email (Spanish)
ID 6 = A/B variant of Email 5

In the root email, if I select the Spanish email for the translation preview, email 6 is not available as a variant. In fact I don't seem to be able to select ID6 at all, with any permutation of dropdowns.

Maybe it is not possible to make a variant of a translated email? Or is it a bug with the implementation?

**Comment by mabumusa1** [comment]
@ts-navghane how can we push this PR across the finish line

**Comment by kuzmany** [comment]
@ts-navghane did you have time to look on reported issues? Thanks,

File: app/bundles/CoreBundle/Form/Type/ContentPreviewSettingsType.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    {
+        $objectId              = $options['objectId'];
+        $translations          = $options['translations'];
+        $variants              = $options['variants'];
+
+        $this->addTranslationOrVariantChoicesElement($builder, self::CHOICE_TYPE_TRANSLATION, $translations, $objectId);
+        $this->addTranslationOrVariantChoicesElement($builder, self::CHOICE_TYPE_VARIANT, $variants, $objectId);
+
+        if ($this->security->isAdmin()
+            || (!$this->security->hasEntityAccess('lead:leads:viewown', 'lead:leads:viewother'))
```
Comment (kuzmany) [inline app/bundles/CoreBundle/Form/Type/ContentPreviewSettingsType.php]: Based on content condition should be without `!`. In other case its hidden for not admin users

**Comment by dadarya0** [comment]
> Looks pretty cool. Works as expected. I tested both email and page. Just If result Is contact without firstname/lastname then return empty row (probably we should display email in the case)
> 
> image

Done

**Review by rohitpavaskar — APPROVED** [review]
LGTM. Thanks!

**Review by kuzmany — APPROVED** [review]
Seems good to me 👍

**Review by Galokh66 — APPROVED** [review]
Tested and verified the preview variations behaviour for emails and landing page. 
Scenarios Tested:

1. Created an Email with landing page, dynamic content and tokens
2. Added ab test variations for the email
3. Added a translation email to it and ab test variation for translated email.
4. Verified preview is working for both emails and its variations
5. Created a “Sales“ user with only view options
6. Verified preview is working for the sales user as well 
7. Opened the preview links on anonymous window and verified it shows permission denied error

1. Created a landing page with some tokens
2. Added ab test variations for the landing page
3.  Added translation page for created page
4. Verified preview is working for both pages and its variations
5. Created a “Sales“ user with only view options
6.  Verified preview is working for the sales user as well 
7. Opened the preview links on anonymous window and verified it shows permission denied error

Tested that contacts without first/last name, show empty option in drop down to select contact for preview

**Comment by dadarya0** [comment]
> MAUT-11257

@Galokh66 please copy and paste tested scenario here instead of ticket number , Jira ticket will not be accessible to other Mautic users.

**Comment by RCheesley** [comment]
@ts-navghane there's no docs for this PR - I think it should be added to this section: https://docs.mautic.org/en/5.x/channels/emails.html#translations to explain that it's possible to preview emails, including translations and A/B variants - WDYT?

cc @mautic/education-team  as this should be in place for the 5.1 release. This section is particularly poor on the docs and could do with some love!

**Comment by RCheesley** [comment]
Docs here: https://github.com/mautic/user-documentation/pull/287

**Pull request merged**